src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/SearchResult.java

```diff
@@ -49,7 +50,7 @@
 
 	public SearchResult(SearchResponse resp) {
 		SearchHits hits = resp.getHits();
-		this.total = hits.getTotalHits();
+		this.total = Optional.ofNullable(hits.getTotalHits()).map(totalHits -> totalHits.value).orElse(0L);
```

Comment:
NP: Could perhaps move this to a utility method since it's been called a few times throughout the code and may be used more in the future.

Reply:
In my opinion, it is the methodology for handling the nullable value and return default value. It is possible to build the common static method and put it into util class, but I don't think it could help more and eventually will make the utils class out of control. any thoughts?

Reply:
Okay, if you don't find it all that helpful then it's fine to leave it as is.